extensions/map-layers-auth/src/ArcGis/ArcGisAccessClient.ts

```diff
@@ -316,28 +328,39 @@ export class ArcGisAccessClient implements MapLayerAccessClient {
       // First attempt: derive the Oauth2 token URL from the 'tokenServicesUrl', exposed by the 'info request'
       try {
         const restUrlFromTokenService = await ArcGisUrl.getRestUrlFromGenerateTokenUrl(urlObj);
+
         if (restUrlFromTokenService === undefined) {
           // We could not derive the token endpoint from 'tokenServicesUrl'.
           // ArcGIS Enterprise Format https://<host>:<port>/<subdirectory>/sharing/rest/oauth2/authorize
-          const regExMatch = url.match(new RegExp(/([^&\/]+)\/rest\/services\/.*/, "i"));
+          const regExMatch = mapLayerUrl.match(new RegExp(/([^&\/]+)\/rest\/services\/.*/, "i"));
           if (regExMatch !== null && regExMatch.length >= 2) {
             const subdirectory = regExMatch[1];
             const port = (urlObj.port !== "80" && urlObj.port !== "443") ? `:${urlObj.port}` : "";
             const newUrlObj = new URL(`${urlObj.protocol}//${urlObj.hostname}${port}/${subdirectory}/sharing/rest/oauth2/${endpointStr}`);
 
             // Check again the URL we just composed
-            return await this.createEndpoint(newUrlObj.toString(), endpointType);
+            const isValidUrl = await this.validateEndpointUrl(newUrlObj.toString());
+            if (isValidUrl === undefined || isValidUrl ) {
```

Comment:
Should this be `undefined || true`?  Wouldn't `isValidUrl === isValidUrl` always be true?

Reply:
Ah sorry I read wrong... should be ok.